src/status_im/contexts/keycard/check/events.cljs

```diff
@@ -23,4 +23,7 @@
                   (do
                     (rf/dispatch [:keycard/disconnect])
                     (rf/dispatch [:open-modal :screen/keycard.empty]))
-                  (rf/dispatch [:keycard/on-application-info-error error])))}]]]})))
+                  (rf/dispatch [:keycard/on-application-info-error
+                                (if (= error :keycard/error.keycard-wrong-profile)
+                                  :keycard/error.keycard-not-empty
+                                  error)])))}]]]})))
```

Comment:
Before login, "This is a different keycard" has higher priority. But after login, "keycard is not empty" has higher priority.

This is all confusing. and complex. @flexsurfer, maybe our current way of global error validation is not working for our requirements.

What do you think? Is it ok to have these error overrides, or should we consider another solution? Any suggestions?

cc:  Thank you @Horupa-Olena for reporting this and starting this discussion

Reply:
We need to find better solution. Instead of stopping at first hard coded error match. Each flow should provide their error priority order to the function itself, and check those errors in that order.
For example, for login case blocked has higher priority over not empty. As mentioned in [this comment](https://github.com/status-im/status-mobile/pull/22094#issuecomment-2674001322)

Reply:
it would be really useful if we could create a table of all possible variants

Reply:
https://www.notion.so/Keycard-scanning-flows-1a48f96fb65c8067a45cef8df38b6cbe

Reply:
so it seems like 1)not keycard, 2) blocked 3)all slots occupied  are general, and should be checked in this order

Reply:
probably 4) should be different card, if profile key-uid is provided, empty/not-empty should be handled outside in every flow

Reply:
So now we have to many issues related to this, honestly i would fix them all in one PR in this task https://github.com/status-im/status-mobile/issues/22166 which im working on